typings/vscode-mssql.d.ts

```diff
@@ -312,6 +312,14 @@ declare module 'vscode-mssql' {
 		generateDeployPlan(packageFilePath: string, databaseName: string, ownerUri: string, taskExecutionMode: TaskExecutionMode): Thenable<GenerateDeployPlanResult>;
 		getOptionsFromProfile(profilePath: string): Thenable<DacFxOptionsResult>;
 		validateStreamingJob(packageFilePath: string, createStreamingJobTsql: string): Thenable<ValidateStreamingJobResult>;
+
+		/**
+		 * Generate sql model from sql script files
+		 * @param projectUri URI of the project file this model is for
+		 * @param modelTargetVersion Target version of SQL Server
+		 * @param filePaths SQL script files of the project
+		 */
+		generateTSqlModel(projectUri: string, modelTargetVersion: string, filePaths: string[]): Thenable<boolean>;
```

Comment:
Have this return a promise (make the implementation async and await the client.sendRequest call)